### PR TITLE
Add testing rule to contribution guidelines

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

@tgstation/commit-access
I'm not super happy with the phrasing here, feedback requested.

Adds a clause to the contributing guidelines that you are expected to have tested your code, which implicitly bans webedited larger PRs.

### CODE REVIEW

**Comment by Iamgoofball** [comment]
im not happy about web edit restrictions but this is a good way to do it that still allows expert users to make tactical web edits

**Comment by Tlaltecuhtli** [comment]
its missing the ' improve not remove ' guide line

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -557,9 +557,11 @@ There is no strict process when it comes to merging pull requests. Pull requests
...
+
 * You are going to be expected to document all your changes in the pull request. Failing to do so will mean delaying it as we will have to question why you made the change. On the other hand, you can speed up the process by making the pull request readable and easy to understand, with diagrams or before/after data. Should you be optimizing a routine you must provide proof by way of profiling that your changes are faster.
 
-* We ask that you use the changelog system to document your change, which prevents our players from being caught unaware by changes - you can find more information about this [on this wiki page](http://tgstation13.org/wiki/Guide_to_Changelogs).
+* We ask that you use the changelog system to document your player facing changes, which prevents our players from being caught unaware by said changes - you can find more information about this [on this wiki page](http://tgstation13.org/wiki/Guide_to_Changelogs).
```
Comment (kriskog) [inline .github/CONTRIBUTING.md:564]: Where do you draw the line here? Obviously github PRs are pointless in the changelog, but keeping a changelog is useful past the players knowing a change was made. I don't see any good reason we cannot log all changes, especially considering the bot uses the changelog tags to label PR's.

Comment (MrDoomBringer) [reply .github/CONTRIBUTING.md:564]: If it's not a player facing change, then the 'changelogs' you're talking about should be descriptive commit messages. Anything else is a hacky workaround for the functionality already provided by git

**Review by Cyberboss — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (kriskog) [reply .github/CONTRIBUTING.md:564]: I mean we can pretend we live in a perfect world, but that doesnt really help us any.

**Review by Rohesie — APPROVED** [review]

**Comment by Jared-Fogle** [comment]
Can you update your PR title to explain what the changes are, not just in your body?

**Review by Mothblocks — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (optimumtact) [reply .github/CONTRIBUTING.md:564]: it's not for us you globalist

**Review by LemonInTheDark — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (SpaceManiac) [reply .github/CONTRIBUTING.md:564]: The primary purpose of the changelog is to inform **players** and **game admins** about recent changes to the game.

In some cases, it makes sense to put a short line mentioning the existence of big refactor in the changelog if the probability that it broke something is high.

Coders and hosts are served by the Git commit log.

**Comment by SpaceManiac** [comment]
> Can you update your PR title to explain what the changes are, not just in your body?

he could save others from the consequences of web editing, but not himself

**Review by SpaceManiac — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (kriskog) [reply .github/CONTRIBUTING.md:564]: Most of our coders are first and foremost players. I think it makes sense to put in the miniscule amount of work that updating the changelog in the case of a refactor or similar is to give the casual coder a summary of what has been merged lately. I doubt most of them even know how to browse the commit log.

Comment (SpaceManiac) [reply .github/CONTRIBUTING.md:564]: > Most of our coders are first and foremost players. 

But only a minority of our players are coders. We shouldn't spam non-coders with irrelevant technical information when what they want to know is whether a bug is fixed or a feature is added.

Comment (kriskog) [reply .github/CONTRIBUTING.md:564]: Players are also submitting issue reports, and giving them a frame of reference to when shit changes can be useful.

But I see you are all against me on this, whatever.

Comment (optimumtact) [reply .github/CONTRIBUTING.md:564]: I will spend them by telling you to stop being such a baby and learn how to use git shortlog

**Review by spookydonut — APPROVED** [review]

**Comment by spookydonut** [comment]
Webedits are allowed if you also have 100% unit test coverage of what you changed

**Review by Twaticus — APPROVED** [review]
choo choo approve train

**Review by stylemistake — APPROVED** [review]

**Review by EOBGames — APPROVED** [review]
All aboard

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -557,9 +557,11 @@ There is no strict process when it comes to merging pull requests. Pull requests
 
 * Make sure your pull request complies to the requirements outlined here
 
+* You are expected to have tested your pull requests if it is anything that would warrant testing. Text only changes, single number balance changes, and similar generally don't need testing, but anything else does. This means by extension web edits are disallowed for larger changes.
```
Comment (MrStonedOne) [inline .github/CONTRIBUTING.md:560]: This is cute.

**Pull request merged**